pkg/ccl/backupccl/restore_planning.go

```diff
@@ -194,19 +194,14 @@ func allocateTableRewrites(
 	// the backup. This generator keeps produced the next descriptor ID.
 	var tempSysDBID sqlbase.ID
 	if descriptorCoverage == tree.AllDescriptors {
-		var err error
-		numberOfIncrements := maxDescIDInBackup - uint32(sql.MaxDefaultDescriptorID)
-		// We need to increment this key this many times rather than settings
-		// it since the interface does not expect it to be set. See client.Inc()
-		// for more information.
-		// TODO(pbardea): Follow up too see if there is a way to just set this
-		//   since for clusters with many descrirptors we'd want to avoid
-		//   incrementing it 10,000+ times.
-		for i := uint32(0); i <= numberOfIncrements; i++ {
-			_, err = sql.GenerateUniqueDescID(ctx, p.ExecCfg().DB)
-			if err != nil {
-				return nil, err
-			}
+		numberOfIncrements := int64(maxDescIDInBackup - uint32(sql.MaxDefaultDescriptorID))
```

Comment:
If we need to lock out full-cluster restore when there are existing descriptors GC'ing or something, we should do that elsewhere, but this code here is responsible for ensuring `keys.DescIDGenerator` is exactly `maxDescIDInBackup` -- i'd feel a little more confident that it was doing that if we just set it directly to that value, of, if we want to inc by a delta, read the existing value instead of _assuming_ it is `sql.MaxDefaultDescriptorID`